### PR TITLE
Bright id 689

### PR DESCRIPTION
**Pull request by TripleSpeeder**
Refactoring of the way channels are shared:

# Goals
- use hostnames instead of ip-address
- enable usage of https protocol
- enable usage of non-standard ports
- as before, channel encryption key should not be stored on channel and only shared via qrcode or deep link
- keep qr-code data at a reasonable size

## Backwards compatibility
As the qrcode format changes, compatibilty is an issue. There is a fallback in place, so new clients will be able to handle old-style qrcodes. But old clients will not understand qrcodes created by new clients.

# Workflow changes
### Channel creator
- creates channel
- uploads well-known file `channelInfo.json` to channel. Example:
```json
{
    "version": 1,
    "type": "GROUP",
    "timestamp": 123456789,
    "ttl": 30000,
    "initiatorProfileId": "abc123456"
}
```
- uploads own profile to channel as usual
- Creates QRCode as a plain url, encoding aesKey and channelID as query parameters. Example:
```http://node.brightid.org/profile/?aes=abc123&id=theChannelId```

### QRcode scanner
- Scans qrcode
- gets aesKey and ChannelID from url parameters
- downloads well-known file `channelInfo.json` from channel to obtain channel parameters
- joins the channel and continue existing connection workflow

### CODE REVIEW

File: BrightID/src/components/PendingConnectionsScreens/actions/channelThunks.js [hunk 1]
```diff
@@ -57,7 +65,9 @@ export const createChannel = (channelType: ChannelType) => async (
     console.log(`Error while crating channel: ${e}`);
```
Comment (adamstallard) [inline BrightID/src/components/PendingConnectionsScreens/actions/channelThunks.js:65]:
```suggestion
    console.log(`Error while creating channel: ${e}`);
```

Comment (adamstallard) [inline BrightID/src/components/PendingConnectionsScreens/actions/channelThunks.js:65]:
```suggestion
    console.log(`Error while creating channel: ${e}`);
```

File: BrightID/src/components/PendingConnectionsScreens/actions/channelThunks.js [hunk 2]
```diff
@@ -189,6 +201,13 @@ export const fetchChannelProfiles = createAsyncThunk(
   async (channelId, { getState, dispatch }) => {
     const channel = selectChannelById(getState(), channelId);
     let profileIds = await channel.api.list(channelId);
+
+    // ignore channelInfo.json
```
Comment (adamstallard) [inline BrightID/src/components/PendingConnectionsScreens/actions/channelThunks.js:205]: `// ignore channelInfo.json`

Could you please explain this comment?

**Review by adamstallard — APPROVED** [review]

File: BrightID/src/components/PendingConnectionsScreens/actions/channelThunks.js (hunk 1 above)
Comment (TripleSpeeder) [reply BrightID/src/components/PendingConnectionsScreens/actions/channelThunks.js:76]: Fixed :)

File: BrightID/src/components/PendingConnectionsScreens/actions/channelThunks.js (hunk 2 above)
Comment (TripleSpeeder) [reply BrightID/src/components/PendingConnectionsScreens/actions/channelThunks.js]: Comment updated

**Pull request merged**